code/game/objects/items/robot/robot_items.dm

```diff
@@ -741,3 +741,193 @@
+			update_icon()
+			return
+	else
+		stored.melee_attack_chain(user, A, params)
+		return
+	. = ..()
+
+/obj/item/borg/apparatus/attackby(obj/item/W, mob/user, params)
+	if(stored)
+		W.melee_attack_chain(user, stored, params)
+		return
+	. = ..()
+
+/////////////////
+//beaker holder//
+/////////////////
+
+/obj/item/borg/apparatus/beaker
+	name = "beaker storage apparatus"
+	desc = "A special apparatus for carrying beakers without spilling the contents. Alt-Z or right-click to drop the beaker."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "borg_beaker_apparatus"
+	storable = list(/obj/item/reagent_containers/glass/beaker,
+				/obj/item/reagent_containers/glass/bottle)
+
+/obj/item/borg/apparatus/beaker/Initialize()
+	. = ..()
+	stored = new /obj/item/reagent_containers/glass/beaker/large(src)
+	RegisterSignal(stored, COMSIG_OBJ_UPDATE_ICON, .proc/update_icon)
+	update_icon()
+
+/obj/item/borg/apparatus/beaker/Destroy()
+	if(stored)
+		var/obj/item/reagent_containers/C = stored
+		C.SplashReagents(get_turf(src))
+		qdel(stored)
+	. = ..()
+
+/obj/item/borg/apparatus/beaker/examine()
+	. = ..()
+	if(stored)
+		var/obj/item/reagent_containers/C = stored
+		. += "The apparatus currently has [C] secured, which contains:"
+		if(length(C.reagents.reagent_list))
+			for(var/datum/reagent/R in C.reagents.reagent_list)
+				. += "[R.volume] units of [R.name]"
+		else
+			. += "Nothing."
+
+/obj/item/borg/apparatus/beaker/update_icon()
+	cut_overlays()
+	if(stored)
+		COMPILE_OVERLAYS(stored)
+		stored.pixel_x = 0
+		stored.pixel_y = 0
+		var/image/img = image("icon"=stored, "layer"=FLOAT_LAYER)
+		var/image/arm = image("icon"="borg_beaker_apparatus_arm", "layer"=FLOAT_LAYER)
+		if(istype(stored, /obj/item/reagent_containers/glass/beaker))
+			arm.pixel_y = arm.pixel_y - 3
+		img.plane = FLOAT_PLANE
+		add_overlay(img)
+		add_overlay(arm)
+	else
+		var/image/arm = image("icon"="borg_beaker_apparatus_arm", "layer"=FLOAT_LAYER)
+		arm.pixel_y = arm.pixel_y - 5
+		add_overlay(arm)
+
+/obj/item/borg/apparatus/beaker/attack_self(mob/living/silicon/robot/user)
+	if(stored && !user.client?.keys_held["Alt"] && user.a_intent != "help")
+		var/obj/item/reagent_containers/C = stored
+		C.SplashReagents(get_turf(user))
+		loc.visible_message("<span class='notice'>[user] spills the contents of the [C] all over the floor.</span>")
+		return
+	. = ..()
+
+/obj/item/borg/apparatus/beaker/extra
+	name = "secondary beaker storage apparatus"
+	desc = "A supplementary beaker storage apparatus."
+
+////////////////////
+//engi part holder//
+////////////////////
+
+/obj/item/borg/apparatus/circuit
+	name = "circuit manipulation apparatus"
+	desc = "A special apparatus for carrying and manipulating circuit boards. Alt-Z or right-click to drop the stored object."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "borg_hardware_apparatus"
+	storable = list(/obj/item/circuitboard,
+				/obj/item/electronics/)

```

Comment:
i dont know but i think that / is useless

Reply:
fixed